components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -118,6 +120,12 @@ open class GleanInternalAPI internal constructor () {
         // This must be set before anything that might trigger the sending of pings.
         initialized = true
 
+        // Process tasks that were queued before init and then clear the list
+        for (task in tasksQueuedBeforeInit) {
```

Comment:
This should probably be moved to `Dispatchers`, maybe `Dispatchers.flushDelayedCalls` or something like that.

Reply:
Totally agree